scriptsOpenMC2GF/base_OpenMC2GF_MGXS.py

```diff
@@ -15,313 +15,281 @@
+    fileString+="\nfeedbackCoeffTStruct TODO;\n" # default: 0
+    fileString+="\nabsoluteDrivelineExpansionCoeff TODO;\n" # default: 0
+    fileString+="\ncontrolRodReactivityMap ( ( 0.1 -0.01 ) ( 0 0 ) ( -0.1 0.01 ) ); \n" 
 
     # Write Number of energy & precursor groups
-    f.write("\ninitPrecursorsLiquidFuel %s;\n\n" % 'true' )
-    f.write("\n energyGroups %i ;\n" % energyGroups)
-    f.write("\n precGroups %i ;\n" % precGroups)
+    fileString+="\ninitPrecursorsLiquidFuel %s;\n\n" % 'true' 
+    fileString+="\n energyGroups %i ;\n" % energyGroups
+    fileString+="\n precGroups %i ;\n" % precGroups
 
     # Write the main nuclear data
+    f.write(fileString)
+    fileString=""
 
     # Add array of region names here, may need another for loop to cycle through the different regions
     OF_NAME = ["hx","intermed","main_fd","pump"]  # The name of the GeN-Foam mesh region this cross section is for
 
-    # Data for GeN-Foam
+    # Data required for GeN-Foam
     fuel_fraction = 1.000000e+00        # The volumetric fuel fraction
-    inv_velocity  = [0, 0, 0, 0, 0, 0]  # m/s
-    diffCoeff     = [0, 0, 0, 0, 0, 0]  # 
-    nusigmaf      = [0, 0, 0, 0, 0, 0]  # 
-    sigmaFissPow  = [0, 0, 0, 0, 0, 0]  # 
-    scattering_P0 = [[0, 0, 0, 0, 0, 0],[0, 0, 0, 0, 0, 0],[0, 0, 0, 0, 0, 0],[0, 0, 0, 0, 0, 0],[0, 0, 0, 0, 0, 0],[0, 0, 0, 0, 0, 0]]   # 
-    scattering    = [0, 0, 0, 0, 0, 0]  # used to calculate sigmaDisapp
-    scattering_ODS= [0, 0, 0, 0, 0, 0]  # used to calculate sigmaDisapp
-    sigmaDisapp   = [0, 0, 0, 0, 0, 0]  # 
-    chi_prompt    = [0, 0, 0, 0, 0, 0]  # 
-    chi_delayed   = [0, 0, 0, 0, 0, 0]  # 
-    beta_eff      = [0, 0, 0, 0, 0, 0]  #
-    decay_const   = [0, 0, 0, 0, 0, 0]  #
-    generic_6     = [1, 1, 1, 1, 1, 1]  # Used for any arrays that are all 1 (discFactor, integralFlux)
-    absorption    = [0, 0, 0, 0, 0, 0]  # 
-    fission       = [0, 0, 0, 0, 0, 0]  # 
-    total         = [0, 0, 0, 0, 0, 0]  # used to calculate sigmaDisapp
+    inv_velocity  = np.zeros(energyGroups)  # s/cm
+    diffCoeff     = np.zeros(energyGroups)  # 
+    nusigmaf      = np.zeros(energyGroups)  # 
+    sigmaFissPow  = np.zeros(energyGroups)  # 
+    scattering_P0 = np.zeros((energyGroups, energyGroups))   # 
+    scattering    = np.zeros(energyGroups)  # used to calculate sigmaDisapp
+    scattering_ODS= np.zeros(energyGroups)  # used to calculate sigmaDisapp
+    sigmaDisapp   = np.zeros(energyGroups)  # 
+    chi_prompt    = np.zeros(energyGroups)  # 
+    chi_delayed   = np.zeros(energyGroups)  # 
+    beta_eff      = np.zeros(energyGroups)  #
+    decay_const   = np.zeros(precGroups)  #
+    generic_6     = np.ones(energyGroups)  # Used for any arrays that are all 1 (discFactor, integralFlux)
+    absorption    = np.zeros(energyGroups)  # 
+    fission       = np.zeros(energyGroups)  # 
+    total         = np.zeros(energyGroups)  # used to calculate sigmaDisapp

```

Comment:
Maybe create a dictionary and initialize each key using some version of
```python
entries_to_change = [...]
for entry in entries_to_change:
    dictionary[entry] = np.zeros(energyGroups)
```

Reply:
For now I will leave this unchanged, but I believe that this will be useful once I implement zone-based MGXS assignment